### DISCUSSION TITLE
How to represent Cisco Voice vLAN in Netbox

### DISCUSSION THREAD

**Discussion by appleoddity** [netbox-community/netbox#13714] [Q&A]
Hi. I am trying to import/represent a large existing network into Netbox. I'm having trouble how to best handle the vLAN configurations. I have a long term vision of using Netbox as the source of truth and pushing config changes to switches from Netbox. My goal right now is to best represent the existing network in Netbox, and then improve on it later.

We have a lot of switch ports that are configured in access mode, but have a voice vLAN assigned:
```
interface GigabitEthernet1/0/36
 switchport access vlan 148
 switchport voice vlan 48
 spanning-tree portfast edge
```

Netbox only offers Access, Tagged, and Tagged (All) modes. The only way I can think of to represent this is to set the mode to Tagged in Netbox, then add an untagged vLAN (for access), and a tagged vLAN (for voice). However, I don't see how this fits well in to automation later on. It would seem to me that I would ultimately have to change the actual port on the switch to be a trunk port with a native vlan, and an allowed vlan list:

```
interface GigabitEthernet1/0/36
 switchport mode trunk
 switchport trunk native vlan 148
 switchport trunk allowed vlan 48, 148
 spanning-tree portfast edge
```

Is this really the best way to do this? Can you offer some hints on how to properly handle this?

**Comment by mtinberg** [comment]
I model it the same way you described, as a tagged port with an untagged vlan for access and the tagged vlan for voice, with boolean a custom_field (tags would work too) enable_voice to indicate this is a voice vlan port not a trunk port so we can template it correctly.  voice vlan is kind of a special case of trunk port as it does pass tagged traffic to the switch built-in to the phone IIUC.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: Appleoddity ***@***.***>
Sent: Thursday, September 7, 2023 1:37 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] How to represent Cisco Voice vLAN in Netbox (Discussion #13714)


Hi. I am trying to import/represent a large existing network into Netbox. I'm having trouble how to best handle the vLAN configurations. I have a long term vision of using Netbox as the source of truth and pushing config changes to switches from Netbox. My goal right now is to best represent the existing network in Netbox, and then improve on it later.

We have a lot of switch ports that are configured in access mode, but have a voice vLAN assigned:

interface GigabitEthernet1/0/36
 switchport access vlan 148
 switchport voice vlan 48
 spanning-tree portfast edge


Netbox only offers Access, Tagged, and Tagged (All) modes. The only way I can think of to represent this is to set the mode to Tagged in Netbox, then add an untagged vLAN (for access), and a tagged vLAN (for voice). However, I don't see how this fits well in to automation later on. It would seem to me that I would ultimately have to change the actual port on the switch to be a trunk port with a native vlan, and an allowed vlan list:

interface GigabitEthernet1/0/36
 switchport mode trunk
 switchport trunk native vlan 148
 switchport trunk allowed vlan 48, 148
 spanning-tree portfast edge


Is this really the best way to do this? Can you offer some hints on how to properly handle this?
